File: Core CS/Algorithms/Implementation/Sales by Match/SalesByMatch.cs
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System;

class Solution {

    static int sockMerchant(int n, int[] ar) {
        return ar.GroupBy(x => x).ToDictionary(x => x.Key, x => x.Count()).Values.Aggregate(0, (acc, x) => acc + x/2);
    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine());

        int[] ar = Array.ConvertAll(Console.ReadLine().Split(' '), arTemp => Convert.ToInt32(arTemp))
        ;
        int result = sockMerchant(n, ar);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Core CS/Algorithms/Implementation/Sales by Match/salesByMatch.scala
import java.io._
import java.text._
import java.util._

object Solution {

    def sockMerchant(n: Int, ar: Array[Int]): Int = {
        return ar.groupBy(identity).mapValues(_.size).foldLeft(0){ case (a, (k, v)) => a + v/2}
    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val n = stdin.readLine.trim.toInt

        val ar = stdin.readLine.split(" ").map(_.trim.toInt)
        val result = sockMerchant(n, ar)

        printWriter.println(result)

        printWriter.close()
    }
}
